api.ts

```diff
@@ -624,7 +623,7 @@ type StringFormulaDefV2<ParamDefsT extends ParamDefs> = BaseFormulaDefV2<ParamDe
 
 type NumericFormulaDefV2<ParamDefsT extends ParamDefs> = BaseFormulaDefV2<ParamDefsT, number> & {
   resultType: ValueType.Number;
-  codaType?: NumberHintTypes;
```

Comment:
As above, this is a breaking change, so we probably need to allow both the `codType` (short-hand) and `schema` (a full schema definition). Bonus points if we can allow exactly one or the other.

Reply:
This is still a breaking change since existing packs are using `codaType` to define a hint type at the top level of the formula definition. So we need to accept both `codaType` and `schema`. You provide `codaType` if you want the current behavior, a simple shorthand if you just want a type hint, but we don't support other attributions of the schema. If you want those, you need to use the `schema` attribute.

To make this either-or crystal clear, we should probably make the typing here something like:

```
BaseFormulaDefV2<ParamDefsT, number> & {
  resultType: ValueType.Number;
  execute(params: ParamValues<ParamDefsT>, context: ExecutionContext): Promise<number> | number;
} & 
  ({schema?: NumberSchema} | {codaType?: NumberHintTypes});
```

It's gnarly that we accept both, but we'd break existing packs if we didn't, and early feedback was that providing `schema` just to give a hint type was too confusing for novice users.

Reply:
Sorry, just getting to these feedbacks. Yeah that makes sense to me. I actually think it'll be fine, now that we're going to make those `ScaleSchema` fields optional.